### PR TITLE
Deduplicate identical typecheck errors between platforms.

### PR DESCRIPTION
**Pull request by rmmh**
**What this PR does / why we need it**: Deduplicate identical typecheck errors between platforms.

This will make typecheck output shorter for code that fails compilation everywhere,
like [this](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/61869/pull-kubernetes-typecheck/3274/).

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by rmmh** [comment]
/assign ixdy

please /approve the sh2ju change :)

**Comment by rmmh** [comment]
Compare:

Initial: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/61869/pull-kubernetes-typecheck/3274/
```
      
      ERROR(windows/amd64) pkg/kubelet/kubelet.go:798:4: names declared but not used
ERROR(darwin/amd64) pkg/kubelet/kubelet.go:798:4: names declared but not used
ERROR(darwin/386) pkg/kubelet/kubelet.go:798:4: names declared but not used
ERROR(windows/386) pkg/kubelet/kubelet.go:798:4: names declared but not used
ERROR(linux/s390x) pkg/kubelet/kubelet.go:798:4: names declared but not used
ERROR(linux/386) pkg/kubelet/kubelet.go:798:4: names declared but not used
ERROR(linux/ppc64le) pkg/kubelet/kubelet.go:798:4: names declared but not used
ERROR(linux/arm64) pkg/kubelet/kubelet.go:798:4: names declared but not used
ERROR(linux/arm) pkg/kubelet/kubelet.go:798:4: names declared but not used
ERROR(linux/amd64) pkg/kubelet/kubelet.go:798:4: names declared but not used
exit status 1
!!! Type Check has failed. This may cause cross platform build failures.
!!! Please see https://git.k8s.io/kubernetes/test/typecheck for more information.
      
      
				
```

Coalesced: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/61872/pull-kubernetes-typecheck/3281/

```
      
      ERROR(all) test/f.go:7:9: no result values expected
ERROR(all) test/f.go:7:9: cannot convert 3 (untyped int constant) to untyped string
ERROR(windows) test/f.go:6:2: Unlink not declared by package unix
exit status 1
!!! Type Check has failed. This may cause cross platform build failures.
!!! Please see https://git.k8s.io/kubernetes/test/typecheck for more information.
      
      
				
```

Coalesced + sh2ju whitespace fix: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/61872/pull-kubernetes-typecheck/3285/
```
ERROR(windows) test/f.go:6:2: Unlink not declared by package unix
ERROR(all) test/f.go:7:9: no result values expected
ERROR(all) test/f.go:7:9: cannot convert 3 (untyped int constant) to untyped string
exit status 1
!!! Type Check has failed. This may cause cross platform build failures.
!!! Please see https://git.k8s.io/kubernetes/test/typecheck for more information.
```

File: test/typecheck/main.go [hunk 1]
```diff
@@ -60,6 +61,8 @@ var (
 		"linux/arm64", "linux/ppc64le",
 		"linux/s390x", "darwin/386",
 	}
+	darwinPlatString  = "darwin/386,darwin/amd64"
+	windowsPlatString = "windows/386,windows/amd64"
```
[the remark this answers is not in the document]
Comment (rmmh) [reply test/typecheck/main.go]: I considered that, but it would take ~20 extra LOC and I don't expect crossPlatforms or these constants to really change.

**Review by BenTheElder — APPROVED** [review]
/lgtm

**Pull request merged**